include/mrdox/Metadata/Template.hpp

```diff
@@ -191,6 +191,36 @@ struct TArg
 
 // ----------------------------------------------------------------
 
+// stores information pertaining to an explicit specialization of a
+// member of an implicitly instantiated class template specialization.
+// this structure is stored in the TemplateInfo corresponding to the
+// outermost specialized template. if the explicitly specialized
+// member is itself a member of a nested template, the SpecializationInfo
+// which stores the arguments for the parent template will additionally
+// store a pointer to the SpecializationInfo for the nested template,
+// recursively. each SpecializationInfo node contains a vector of
+// SymbolIDs pairs `(specialized, primary)`, where `specialized` is the
+// replacement definition of `primary` for the given set of template
+// arguments of its parent template(s).
```

Comment:
ugh this is a really ugly comment can we please use /* */ for multiline exposition and also this should be a javadoc if it is going to be in a public API:
```
/** Metadata for explicit specializations of certain instantiated class template members

    This stores information pertaining to an explicit specialization of a
    member of an implicitly instantiated class template specialization.
    this structure is stored in the `TemplateInfo` corresponding to the
    outermost specialized template. if the explicitly specialized
    member is itself a member of a nested template, the `SpecializationInfo`
    which stores the arguments for the parent template will additionally
    store a pointer to the `SpecializationInfo` for the nested template,
    recursively. each `SpecializationInfo` node contains a vector of
    `SymbolID` pairs (specialized, primary), where `specialized` is the
    replacement definition of `primary` for the given set of template
    arguments of its parent template(s).
*/
struct SpecializationInfo;
```